internal/otel/providers/hcp/provider.go

```diff
@@ -95,9 +95,15 @@ func (m *hcpProvider) Retrieve(
 		return nil, err
 	}
 
-	// 3. B: Build external pipeline
+	// 3. B: Build external pipeline We need to build this external pipeline because of how otel merges configuration.
```

Comment:
Any important  doc links on this 🤔 im not sure I fully understand this need to double the config yet, I have an idea but trying to dig a bit deeper, particularly this part " The external generator will ensure that the forward component.ID and other components are included in the service stanza"  😓

Reply:
We end up creating two pipelines here, one for the external metric pipeline and one for the HCP Pipeline. In order to generate the external Pipeline we call the external builder as well.